nsxt/resource_nsxt_edge_cluster.go

```diff
@@ -0,0 +1,309 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+func resourceNsxtEdgeCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtEdgeClusterCreate,
+		Read:   resourceNsxtEdgeClusterRead,
+		Update: resourceNsxtEdgeClusterUpdate,
+		Delete: resourceNsxtEdgeClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"revision":     getRevisionSchema(),
+			"description":  getDescriptionSchema(),
+			"display_name": getDisplayNameSchema(),
+			"tag":          getTagsSchema(),
+			"edge_ha_profile_id": {
```

Comment:
Is it possible to add multiple ha profiles? Same about below

Reply:
This is odd - the API has a list there and there are two types (bridge, edge). Yet I can't create a bridge profile using curl (server returns 201 but nothing is created), and UI lets me select only one profile.
Not sure what we do with this (nor where we gather more info).

Reply:
Looks like only single HA profile can be assigned - hence TypeList is not needed